bson/src/main/org/bson/codecs/pojo/PojoCodecImpl.java

```diff
@@ -266,9 +267,15 @@ private void decodeProperties(final BsonReader reader, final DecoderContext deco
         return codec;
     }
 
+    @SuppressWarnings("unchecked")
     private <S> Codec<S> getCodecFromPropertyRegistry(final PropertyModel<S> propertyModel) {
         try {
-            return propertyCodecRegistry.get(propertyModel.getTypeData());
+            Codec<S> codec = propertyCodecRegistry.get(propertyModel.getTypeData());
+            BsonType representation = propertyModel.getBsonRepresentation();
+            if (codec instanceof RepresentationConfigurable && representation != null) {
```

Comment:
This could be a good place to check for user bsonRepresentation errors?

Reply:
Yep added a check here to throw errors if a BsonRepresentation is provided but the codec isn't RepresentationConfigurable. The rest of the checks are done in the `withRepresentation` methods of the codecs that are RepresentationConfigurable.